src/sentry/api/serializers/rest_framework/rule.py

```diff
@@ -51,13 +51,25 @@ class RuleSerializer(serializers.Serializer):
     )
     frequency = serializers.IntegerField(min_value=5, max_value=60 * 24 * 30)
 
+    def validate_environment(self, attrs, source):
```

Comment:
This method is super weird and feels bad, I think there might/must be a better way to do it but I'm not familiar enough with DRF to really know. The result from L60 ends up being cast back to a `unicode` type at some point (???) so this returns a numeric value, casts it to a unicode value, then casts it back again to a numeric when it actually gets saved to the database. 🤔

Reply:
That happens here apparently: https://github.com/encode/django-rest-framework/blob/647fdb51d203976a82a9cde1c7a535bd89c73135/rest_framework/serializers.py#L385-L386

Reply:
I guess the options here are:

1. Hold my nose and just use this as is, as gross and weird as it is (I can't think of cases where this would break, but it seems unnecessarily convoluted),
2. Store the `environment` object as an instance attribute after this method has been executed (this solves the type weirdness, but I hate "this method must be run before you can run this other method" types of dependencies)
3. Write a custom field that actually returns the `Environment` object (I really don't want to do this, and not even sure how realistic it is since it needs access to `self.context['project']`)

Reply:
Could use some feedback here, otherwise I'm just going to rebase this and regenerate the migration and keep moving forward as is